src/resolve/resolved-dns-rr.c

```diff
@@ -826,6 +827,115 @@ static char *format_txt(DnsTxtItem *first) {
         return s;
 }
 
+static char *format_svc_param_value(DnsSvcParam *i) {
+        char *value = NULL;
+        int r;
+
+        switch (i->key) {
+        case DNS_SVC_PARAM_KEY_ALPN: {
+                size_t sz = 0, offset = 0;
+                _cleanup_strv_free_ char **values_strv = NULL;
+                while (offset < i->length) {
+                        char *alpn;
+                        sz = (uint8_t) i->value[offset++];
+                        alpn = strndup((char *)&i->value[offset], sz);
+                        if (!alpn)
+                                return NULL;
+                        strv_push(&values_strv, alpn);
+                        offset += sz;
+                }
+                value = strv_join(values_strv, ",");
+                break;
+        }
+        case DNS_SVC_PARAM_KEY_PORT: {
+                uint16_t port = unaligned_read_be16(i->value);
+                r = asprintf(&value, "%" PRIu16, port);
+                if (r < 0)
+                        return NULL;
+                break;
+        }
+        case DNS_SVC_PARAM_KEY_IPV4HINT: {
+                const struct in_addr *addrs = (struct in_addr*) &i->value;
```

Comment:
is this guaranteed to be properly aligned?

Reply:
Oh hm, I don't think so...

Reply:
Ah, so, it actually is aligned. It's sort of by mistake though since it is just a result of the value member following the list pointers in the struct layout, and those have a greater alignment requirement.

Since it already works, I've just declared it (DnsSvcParam.value) to have the necessary alignment to make it explicit. Is that alright?